067-kraft-jbod-support.md

```diff
@@ -0,0 +1,224 @@
+There is currently no way how to _programmatically_ move the metadata log to another directory similarly to moving partition replicas.
+While such feature might be added in the future, this proposal does not require it.
+
+#### Strimzi challenges
+
+The JBOD support in KRaft-based Kafka clusters provides several challenges that are addressed in this proposal:
+* What level of control should the users have over the placement of the metadata log?
+* How does Strimzi move or prevent moving of the metadata log between directories?
+* Should Strimzi support dedicated metadata directories?
+* Should Strimzi support JBOD storage in controller-only nodes?
+
+These challenges are addressed in the next sections.
+
+## Proposal
+
+### API changes
+
+A new optional property named `kraftMetadata` will be added to the volume definition in the Strimzi `Kafka` custom resource API.
+This property will have an Enum value and the supported values will be:
+* `null` (or not set)
+* `shared`
+
+The volume marked with `kraftMetadata: shared` will be used to store the metadata log.
+In a `type: jbod` storage configuration, only one volume will be permitted to have the `kraftMetadata` property set to `shared`.
+This will be validated in the Cluster Operator and when it is set for multiple volumes at the same time, an `InvalidResourceException` will be thrown.
+It is also possible to not set this option for any volume as well.
+In that case, the operator will decide which volume should be used for the metadata log on its own.
+
+Following example shows how the configuration might look like in order to use the second volume to store the metadata log:
+```yaml
+apiVersion: kafka.strimzi.io/v1beta2
+kind: KafkaNodePool
+metadata:
+  name: broker
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - broker
+    storage:
+      type: jbod
+      volumes:
+      - id: 0
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+      - id: 1
+        type: persistent-claim
+        size: 100Gi
+        deleteClaim: false
+        kraftMetadata: shared
+```
+
+The reason why the `kraftMetadata` property is not just a boolean is to make it possible to extend the support in the future.
+One such possibility would be to support dedicated metadata volumes, which is discussed in a separate section.
+
+The `kraftMetadata` option will be ignored when used with ZooKeeper-based clusters.
+
+### Kafka configuration
+
+Currently, we only use the `log.dirs` option to configure the log directories.
+With this proposal, the `log.dirs` option will be configured in the same way as present.
+In addition to that, we will also configure the `metadata.log.dir` option that we do not use today.
+This option will be configured only for KRaft clusters.
+It will be set to the volume marked as `kraftMetadata: shared`.
```

Comment:
It's not clear to me why the "shared" naming. It sounds like you are sharing the volume of the metadata with something else. But what? Isn't it dedicated to host the metadata only?

Reply:
Actually I forgot that even brokers store metadata as "observers" so it makes sense.